lib/org/chromium/webidl/ParserRunner.js

```diff
@@ -8,45 +8,78 @@ foam.CLASS({
   name: 'ParserRunner',
   extends: 'org.chromium.webidl.PipelineRunner',
 
-  requires: ['org.chromium.webidl.BaseParser'],
+  requires: [
+    'org.chromium.webidl.BaseIDLFile',
+    'org.chromium.webidl.BaseParser',
+    'org.chromium.webidl.ast.Definition',
+    'org.chromium.webidl.Parser as DefaultParser',
+  ],
 
   documentation: 'Runnable Box that parsers IDLFileContents into ASTs.',
 
-  methods: [
-    function run(message) {
-      if (this.validateMessage(message)) return;
+  properties: [
+    {
+      class: 'Class',
+      documentation: 'Class of the parser used to parse the IDL files.',
+      name: 'Parser',
+      factory: function() {
+        // TODO: Replace with a smarter method of choosing a parser.
+        return this.DefaultParser;
+      },
+    },
+    {
+      class: 'String',
+      name: 'ioRelationshipType',
+      documentation: 'The n:m relationship type of input-to-output.',
+      value: '1:many',
+    },
+    {
+      class: 'Class',
+      documentation: 'Type of input parameter of run().',
+      name: 'inputType',
```

Comment:
We should probably have all this class's `class: 'Class'` properties be named the same: either suffix of `Type` or suffix of `Cls`.